SDAPackage/SDA/obstacle_map.py

```diff
@@ -27,11 +27,56 @@ def add_obstacle(self, obstacle_to_add):
         :param obstacle_to_add: The obstacle to add to the map
         :type obstacle_to_add: StationaryObstacle
         """
+        final_obstacles = np.array([])
         if self.obstacles.size != 0:
-            self.obstacles = np.hstack((self.obstacles, obstacle_to_add))
+            final_obstacles = np.hstack((self.obstacles, obstacle_to_add))
+            final_obstacles = overlapping_obstacles(final_obstacles)
+            self.obstacles = np.hstack((self.obstacles, final_obstacles))
         else:
             self.obstacles = np.array([obstacle_to_add])
 
+    def overlapping_obstacles(self, final_obstacles):
+        for first_obstacle_index in range(len(final_obstacles)):
+            for second_obstacle_index in range(first_obstacle_index + 1, len(final_obstacles)):
+                if (overlap_obstancles_boolean(first_obstacle_index, second_obstacle_index, final_obstacles) is False)
```

Comment:
I'd recommend changing the ```overlap_obstancles_boolean``` to instead be something like ```do_obstacles_overlap()```. You also need a ```self.``` in front of the method call

Reply:
Also, a few other issues:

1) You appear to pass the indexes and the final obstacles array. It might be easier just to pass the two obstacles as parameters. That way, you reduce the amount of code you need to write and you make the method more clear
2) The assertion here is backwards. If the obstacles do overlap, then you want to execute the code inside the if statement. If they do not overlap, then you do not